src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs

```diff
@@ -1,11 +1,47 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class ICompilationExtensions
     {
+        public static ImmutableArray<Compilation> GetReferencedCompilations(this Compilation compilation)
+        {
+            var builder = ArrayBuilder<Compilation>.GetInstance();
```

Comment:
Even 1/3rd of the consumers needing updates is enough in my book to mean we should consider making the API public. Should we file a bug to track further?